src/cache.rs

```diff
@@ -179,19 +179,22 @@ impl Cache {
     }
 
     /// Look up custom patch (<name>.patch). If it exists, store it in a variable.
-    fn find_patch(patch_name: &str, custom_pages_dir: Option<&Path>) -> Option<PathBuf> {
-        custom_pages_dir
-            .map(|custom_dir| custom_dir.join(patch_name))
-            .filter(|path| path.exists() && path.is_file())
+    fn find_patch(patch_name: &str, custom_pages_dir: impl AsRef<Path>) -> Option<PathBuf> {
+        let patch_path = custom_pages_dir.as_ref().join(patch_name);
+        if patch_path.exists() && patch_path.is_file() {
+            return Some(patch_path);
+        }
+        None
```

Comment:
Since we will allocate a `PathBuf` here in every case with `Path::join`, it would be good practice to just take a `PathBuf` upfront so that if you already have the custom pages directory as a `PathBuf` and don't need it anymore, you can give the memory to the function. I personally liked the `Option::filter` version better, but you can decide which one you like best:`Some(custom_pages_dir.join(patch_name)).filter(|path| path.is_file())` (`is_file` also implies `exists`).

Reply:
Yeah I like the 1 liner too. I didn't think of manually wrapping it in Some to get access to the filter method again.